pages/index.js

```diff
@@ -1,19 +1,78 @@
 import ProductsList from "@/components/ProductsList";
 import Icon from "@/components/Icons";
 import { StyledCreateLink } from "@/components/Buttons";
+import styled from "styled-components";
+import { useEffect, useState } from "react";
+
+const StyledSearchForm = styled.form`
+  display: flex;
+  margin-block-start: 30px;
+  border-radius: 5px;
+  width: 100%;
+  border: 1px solid;
+  box-shadow: inset 0px 0px 2px var(--shadowColor);
+  padding-inline: 10px;
+  background-color: white;
+  &:focus-within {
+    border: 2px solid var(--accentColor);
+  }
+`;
+
+const StyledSearchInput = styled.input`
+  background: transparent;
+  border: none;
+  outline: none;
+  padding: 10px;
+  width: 100%;
+  font-size: 16px;
+`;
 
 export default function HomePage({
   products,
   onDeleteProduct,
   onToggleShoppingList,
 }) {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [foundProducts, setFoundProducts] = useState();
+  useEffect(() => {
+    setFoundProducts(
+      products.filter((product) =>
+        product.name.toLowerCase().includes(searchTerm.toLowerCase())
+      )
+    );
+  }, [searchTerm, products]);
+
   return (
     <main>
+      <StyledSearchForm>
+        <Icon variant="search" />
+        <StyledSearchInput
+          onChange={(event) => setSearchTerm(event.target.value)}
+          type="text"
+          id="searchTerm"
+          name="searchTerm"
+          placeholder="Search a product"
+        ></StyledSearchInput>
+      </StyledSearchForm>
```

Comment:
You could think about moving your search form into its own component. Maybe you want to reuse it some day to search for stores as well? But maybe it's too early for this change now - I'm not quite sure.